### PR TITLE
Gearama/continuation

### PR DESCRIPTION
**Pull request by gearama**
updated to the latest nd greatest agents , logging now up and runnign

### CODE REVIEW

File: agents/logsuploader/main.go [hunk 1]
```diff
@@ -1,27 +1,105 @@
...
-	var name string
-	client := appinsights.NewTelemetryClient("320dcf98-173f-429b-ab39-df8b4951fb94")
-	metric := appinsights.NewMetricTelemetry("Queue length", 1)
-	event := appinsights.NewEventTelemetry("event")
-	metric.Properties["Queue name"] = "queuename"
-	client.Track(metric)
-	client.Track(event)
+	fmt.Println("Starting logs uploader.")
+
+	if checkFirstInstance() == false {
```
Comment (tokaplan) [inline agents/logsuploader/main.go:18]: Can we not quit, but rather sleep in a loop and check if it's running every couple of minutes, and start it if it failed?

File: agents/logsuploader/uploader.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	fmt.Println("attempting to parse and load", maybeJSON)
+	result := false
+
+	var deserialized Event
+
+	err := json.Unmarshal([]byte(maybeJSON), &deserialized)
+
+	if err == nil {
+		event := createEvent(deserialized)
+		client.Track(event)
```
Comment (tokaplan) [inline agents/logsuploader/uploader.go:21]: Is there client.Flush()? Let's make an effort to send it right away if available.

Comment (gearama) [reply agents/logsuploader/uploader.go:21]: done

File: agents/logsuploader/main.go (hunk 1 above)
Comment (gearama) [reply agents/logsuploader/main.go:18]: we could , i want to minimize resource use , thus exit so we only have one instance up and running

Comment (tokaplan) [reply agents/logsuploader/main.go:18]: I'd say do it. As long as it only wakes up maybe once a minute - it's worth it. And if possible to log from here - log an error message saying the spawned process disappeared, that way we'll know if this is a persistent problem.

**Pull request merged**